src/main/java/it/gov/pagopa/reward/notification/repository/RewardOrganizationExportsRepositoryExtendedImpl.java

```diff
@@ -23,6 +29,89 @@ public RewardOrganizationExportsRepositoryExtendedImpl(ReactiveMongoTemplate mon
         this.mongoTemplate = mongoTemplate;
     }
 
+    @Override
+    public Flux<RewardOrganizationExport> findAllBy(String organizationId, String initiativeId, Pageable pageable, ExportFilter filters) {
+
+        Criteria criteria = Criteria
+                .where(RewardOrganizationExport.Fields.organizationId).is(organizationId)
+                .and(RewardOrganizationExport.Fields.initiativeId).is(initiativeId);
+
+        // if filters are set, update the criteria; else, use default query
+        updateCriteriaWithFilters(criteria, filters);
+
+        if (filters != null && filters.getStatus() != null && checkStatusNotInExported(filters.getStatus())) {
+            return Flux.empty();
+        } else {
+            return mongoTemplate
+                    .find(
+                            Query.query(criteria).with(getPageable(pageable)),
+                            RewardOrganizationExport.class
+                    );
+        }
+    }
+
+    @Override
+    public Mono<Long> countAll(String organizationId, String initiativeId, ExportFilter filters) {
+
+        Criteria criteria = Criteria
+                .where(RewardOrganizationExport.Fields.organizationId).is(organizationId)
+                .and(RewardOrganizationExport.Fields.initiativeId).is(initiativeId);
+
+        // if filters are set, update the criteria; else, use default query
+        updateCriteriaWithFilters(criteria, filters);
+
+        if (filters != null && filters.getStatus() != null && checkStatusNotInExported(filters.getStatus())) {
+            return Mono.empty();
```

Comment:
count should always return a value
```suggestion
            return Mono.just(0L);
```